language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/definition/LSStandardLibCache.java

```diff
@@ -0,0 +1,233 @@
+import org.ballerinalang.model.tree.TopLevelNode;
+import org.ballerinalang.util.diagnostic.DiagnosticListener;
+import org.wso2.ballerinalang.compiler.Compiler;
+import org.wso2.ballerinalang.compiler.FileSystemProjectDirectory;
+import org.wso2.ballerinalang.compiler.SourceDirectory;
+import org.wso2.ballerinalang.compiler.tree.BLangAnnotation;
+import org.wso2.ballerinalang.compiler.tree.BLangFunction;
+import org.wso2.ballerinalang.compiler.tree.BLangImportPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.tree.BLangTypeDefinition;
+import org.wso2.ballerinalang.compiler.tree.expressions.BLangConstant;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.CompilerOptions;
+import org.wso2.ballerinalang.compiler.util.ProjectDirConstants;
+import org.wso2.ballerinalang.compiler.util.diagnotic.DefaultDiagnosticListener;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import static org.ballerinalang.compiler.CompilerOptionName.COMPILER_PHASE;
+import static org.ballerinalang.compiler.CompilerOptionName.OFFLINE;
+import static org.ballerinalang.compiler.CompilerOptionName.PRESERVE_WHITESPACE;
+import static org.ballerinalang.compiler.CompilerOptionName.PROJECT_DIR;
+
+/**
+ * Standard Library Cache Implementation for the Language Server.
+ * This will be used by the Goto Definition and the Reference Feature Implementations
+ *
+ * @since 1.2.0
+ */
+public class LSStandardLibCache {
+    private static final LSStandardLibCache STANDARD_LIB_CACHE;
+    private static final ReentrantLock lock = new ReentrantLock();
+    private static final String CACHE_PROJECT_NAME = "stdlib_cache_proj";
+    private static final String TOML_CONTENT = "[project]\norg-name= \"lsbalorg\"\nversion=\"1.1.0\"\n\n[dependencies]";
+    private Path stdlibCacheProjectPath;
+    private Path stdlibSourceRoot;
+    private boolean cacheProjectRootSuccess;
+    private LoadingCache<String, List<TopLevelNode>> topLevelNodeCache;
+
+    static {
+        STANDARD_LIB_CACHE = new LSStandardLibCache();
+    }
+
+    private LSStandardLibCache() {
+        topLevelNodeCache = CacheBuilder.newBuilder()
+                .maximumSize(25)
+                .expireAfterAccess(30, TimeUnit.MINUTES)
+                .build(new CacheLoader<String, List<TopLevelNode>>() {
+                    @Override
+                    public List<TopLevelNode> load(@Nonnull String module) throws UnsupportedEncodingException {
+                        return getNodesForModule(module);
```

Comment:
Shall we have a trace log here stating that "module x is loading into cache...." similar to https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/LSModuleCompiler.java#L287-L292. Will be useful to identify if there's never resolving module imports.